packages/lit-dev-content/samples/examples/react-basics/deps.ts

```diff
@@ -0,0 +1,31 @@
+/*
+  Dependencies for examples include:
+  - React
+  - ReactDOM
+  - SimpleGreeting web component
+*/
+
+import {html, LitElement} from 'lit';
+import {customElement, property} from 'lit/decorators.js';
+
+declare global {
+  interface Window {
+      React: any;
+      ReactDOM: any;
+  }
+}
+
+const React = window.React;
+const ReactDOM = window.ReactDOM;
```

Comment:
i feel like this is a little too different from what a react user would expect.
The example leverages the fact that `react` is on the `window` object.

Most users would expect some kind of `import * as React from 'react'` or similar.

I think faking the module imports and declaring a namespace does the same as your example but still retains the ergonomics of "modern" best practices

Reply:
Modern react actually moved away from needing to import `React` for JSX use. https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html

Though users would have to import `React` to pass into `createComponent()` or for hooks.

I do agree that doing `declare var` is not very sensical. Perhaps some comments in the deps file that explains why it's being done that way would be nice?

Reply:
So I think with this one we should try and emulate other sites like codesandbox.

Or should we? I wanna use the new stuff. But is the new stuff too new though ;_;

Reply:
I think we'll have to still have the import in our playground since the new jsx transform is more for tools.

I did find that instead of adding the umd cdn from unpkg in the html, we can use this esm cdn.
```js
import React from 'https://esm.sh/react@18';
import ReactDOM from 'https://esm.sh/react-dom@18/client';
```
Unfortunately typescript will error since it can't find types. `// @ts-ignore` can be added to each line to suppress it, but it does work without it too.

https://lit.dev/playground/#gist=b3747690a33af6e883d456fbaa00fec0

Reply:
Alternatively, we can add import maps to our playground `project.json` like
```json
  "importMap": {
    "imports": {
      "react": "https://esm.sh/react@18",
      "react-dom/": "https://esm.sh/react-dom@18/"
    }
  }
```
as documented here https://github.com/google/playground-elements#import-maps

This would allow straight up
```js
import React from 'react';
import ReactDOM from 'react-dom/client';
```
within the playground ide.

It does require we add `https://esm.sh/*` to the csp connect-src here
https://github.com/lit/lit.dev/blob/e85103f767e113749f76fe291dfb740fdceae608/packages/lit-dev-server/src/middleware/content-security-policy-middleware.ts#L191

There's no way for a user to recreate this in a custom playground example though as they won't be able to modify the `project.json`.

Reply:
Definitely an option, I think I started here but needed to bypass for some reason during the build.

Thanks for the suggestion! Worth exploring

Reply:
> There's no way for a user to recreate this in a custom playground example though as they won't be able to modify the project.json.

Oh interesting! didn't knnow that. I thought you could access the `+ project.json` by creating the file, then it copies. Did we lose that behavior or was it a fever dream? lol

Reply:
Users can create a `package.json` to add dependencies, which playground-elements then fetches from unpkg.com.

Reply:
Makes sense

Reply:
I've isolated React into it's own module. I think that's a good checkpoint for this PR and finding a durable solution (perhaps with types?) in a future PR. I'll file a bug.